src/test/java/bio/terra/common/TestUtils.java

```diff
@@ -346,4 +347,19 @@ public static void assertError(
     Throwable throwable = assertThrows("expect a failure", expectedThrowable, runnable);
     assertThat(throwable.getMessage(), containsString(expectedMessage));
   }
+
+  /**
+   * Given an object with a private field, extract the value of that field and return it. Note: this
+   * should not be used for classes that we control but for third party library classes that aren't
+   * easily mocked (e.g. certain Google client SDK configuration classes)
+   *
+   * @param object The object to extract the value from
+   * @param fieldName The name of the field to extract
+   * @return A typed object containing the value of the field
+   * @param <T> The expected type of the field
+   */
+  public static <T> T extractValueFromPrivateObject(Object object, String fieldName) {
+    return objectMapper.convertValue(
+        ReflectionTestUtils.getField(object, fieldName), new TypeReference<>() {});
+  }
```

Comment:
What is `objectMapper` doing here? You shouldn't need to do any conversion other than a cast after calling `getField()`. You'll get an unchecked warning but that's expected when using reflection. I'd use an annotation to suppress the warning.
```suggestion
  @SuppressWarnings("unchecked")
  public static <T> T extractValueFromPrivateObject(Object object, String fieldName) {
    return (T) ReflectionTestUtils.getField(object, fieldName);
  }
```

Reply:
Ah I hate that.  I actually prefer the object mapper approach

Reply:
The mapper approach won't work for all `T`. It round trips `object` JSON and then back to type `T`, so it only works if `T` can be serialized as a json object. Since `object` is already type `T` this is wasteful and, in some cases, incorrect.

Reply:
If you don't like the `@SuppressWarnings`, you can't avoid it -- `ObjectMapper.convertObject()` is doing it too, so the suppress is still there, if not directly in your code.